File: backend/src/main/java/com/gestion/reservation_terrain/model/Paiement.java
package com.gestion.reservation_terrain.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.sql.Date;
import java.util.UUID;

@Data
@Entity
@Table(name = "paiement")
public class Paiement {
    @Id
    private UUID uuid;

    private Float montant;
    private Date datePaiement;
    private String numeroCarte;
    private String dateExpiration;
    private String codeVerification;
    private String titulaireCarte;
    private String typeCarte;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "uuid")
    @JsonIgnore
    private Reservation reservation;

}

File: frontend/src/app/models/paiement.ts
import {Reservation} from "./reservation";

export interface Paiement {
  uuid: string;
  montant: number;
  datePaiement?: Date;
  numeroCarte?: string;
  dateExpiration?: string;
  codeVerification?: string;
  titulaireCarte?: string;
  typeCarte?: string;
  reservation?: Reservation;
}
